src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/Utils.cs

```diff
@@ -723,52 +723,17 @@ internal static X509Certificate2Collection BuildBagOfCerts(KeyInfoX509Data keyIn
 
         internal static string EncodeHexString(byte[] sArray)
         {
-            return EncodeHexString(sArray, 0, (uint)sArray.Length);
-        }
-
-        internal static string EncodeHexString(byte[] sArray, uint start, uint end)
-        {
-            string result = null;
-            if (sArray != null)
-            {
-                char[] hexOrder = new char[(end - start) * 2];
-                for (uint i = start, j = 0; i < end; i++)
-                {
-                    int digit = sArray[i];
-                    hexOrder[j++] = HexConverter.ToCharUpper(digit >> 4);
-                    hexOrder[j++] = HexConverter.ToCharUpper(digit);
-                }
-                result = new string(hexOrder);
-            }
-            return result;
+            return HexConverter.ToString(sArray);
         }
 
         internal static byte[] DecodeHexString(string s)
         {
             string hexString = Utils.DiscardWhiteSpaces(s);
-            uint cbHex = (uint)hexString.Length / 2;
-            byte[] hex = new byte[cbHex];
-            int i = 0;
-            for (int index = 0; index < cbHex; index++)
-            {
-                hex[index] = (byte)((HexToByte(hexString[i]) << 4) | HexToByte(hexString[i + 1]));
-                i += 2;
-            }
+            byte[] hex = new byte[hexString.Length / 2];
+            HexConverter.TryDecodeFromUtf16(hexString.AsSpan(0, hex.Length * 2), hex);
```

Comment:
This isn't an equivalent replacement.  The current behavior turns non-hex characters into 0xFF (messing up the byte they appear in) but moves on, this changes it to stop at the first error.

HexToByte can be replaced with the common function, but the loop that doesn't look for errors should be restored.

Reply:
Reverted this part while re-using the common function in de2085b.
Good catch!